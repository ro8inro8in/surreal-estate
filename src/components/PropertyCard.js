import React from "react";
import PropTypes from "prop-types";


const PropertyCard = (props) => {
  const { bathrooms, bedrooms, city, email, price, title, type } = props;
  return (
    <div className="PropertyCard">
      <div className="bathrooms">
        bathrooms:
        {bathrooms.max}
      </div>
      <div className="bedrooms">
        bedrooms:
        {bedrooms.max}
      </div>
      <div className="city">
        city:
        {city}
      </div>
      <div className="email">
        email:
        {email}
      </div>
      <div className="price">
        price:
        {price.max}
      </div>
      <div className="title">
        title:
        {title}
      </div>
      <div className="type">
        type:
        {type}
      </div>
    </div>
  );
};

export default PropertyCard;

PropertyCard.propTypes = {
  bathrooms: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
  bedrooms: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};
