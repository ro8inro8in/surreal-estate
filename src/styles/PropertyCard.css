.property-card {
  margin-top: 5%;
  margin-bottom: -5%;
  background-color: rgb(255, 254, 254);
  box-shadow: 0 8px 16px rgb(38 38 48 / 20%);
  padding: 3%;
  text-align: center;
  border-radius: 8px;
}
.card-btn {
  display: flex;
  justify-content: start;
  width: 100%;
  height: 25px;
  margin-top: 20px;
  border-radius: 4px;
  background: #b462b8;
  color: white;
  border: 0px transparent;
  text-align: center;
  display: inline-block;
  cursor: pointer;
  box-shadow: 0 8px 16px rgb(38 38 48 / 20%);
}
.card-btn:hover {
  background-color: rgba(189, 23, 235, 0.782);
  border-radius: 4px;
  transition: all 0.2s ease out;
}

.property-card-title {
  font-size: 36px;
}
.property-card-type {
  font-size: 30px;
}
.property-card-city{
  font-size: 26px;
}

.property-card-bathrooms {
  color: pink;
  font-size: 26px;
  
}
fa-bath{
  color: rgb(0, 255, 51);
  font-size: 26px;
}

.property-card-bedrooms {
  color: red;
  font-size: 26px;
}

.property-card-price {
  color: green;
  font-size: 26px;
}

.property-card-email {
  color: blue;
  font-size: 26px;
}