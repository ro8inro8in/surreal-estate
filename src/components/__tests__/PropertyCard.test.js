import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "",
    type: "Flat",
    bathrooms: {
      min: 1,
      max: 5,
    },
    bedrooms: {
      min: 1,
      max: 5,
    },
    price: 666,
    city: "Manchester",
    email: "",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

