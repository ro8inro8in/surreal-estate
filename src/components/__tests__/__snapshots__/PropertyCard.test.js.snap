// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="PropertyCard"
  >
    <div
      class="bathrooms"
    >
      bathrooms:5
    </div>
    <div
      class="bedrooms"
    >
      bedrooms:5
    </div>
    <div
      class="city"
    >
      city:Manchester
    </div>
    <div
      class="email"
    >
      email:
    </div>
    <div
      class="price"
    >
      price:
    </div>
    <div
      class="title"
    >
      title:
    </div>
    <div
      class="type"
    >
      type:Flat
    </div>
  </div>
</DocumentFragment>
`;
